### PR TITLE
Remove AspNet 2.2 dependencies

### PR DESCRIPTION
**Pull request by d-barker**
I have been investigating a VS Code issue where the AspNet.MVC Analyzer fails with Null pointer exception, the cause of this is a reference to a 2.2 version of AspNet.MVC (and other related libraries). In order to clean out the house, I have had to change this package to use the Core version of Identity.

As a result this could be a breaking change, as the .AddDefaultTokenProviders() method is no longer called by default and is now the responsibility of the caller to add the token providers. The call to ConfigureMongoDbIdentity now returns an IdentityBuilder so the caller can use this reference to make additional calls to add the token providers or to call .AddDefaultTokenProviders().

The AddDefaultTokenProviders is part of the netcoreapp3.1 framework (MS moved away from a seperate package) and as a result makes this library dependant on netcoreapp3.1. I use this package in a number of libraries which are netstandard2.1 and as a result I don't want to have to upgrade everything to netcoreapp3.1.

So this pull request does the following:
1. Removes the AspNet 2.2 references
2. Adds a reference to the Microsoft.Extensions.Identity.Core
3. Changed the method ConfigureMongoDbIdentity to use .AddIdentityCore<>() and .AddRoles<>() as two seperate calls instead of the single that is shipped with AspNet 3.1
4. Updated unit tests to make the call to .AddDefaultTokenProviders() where applicable (all tests are passing with the changes, I had to fix one of the unit tests where a new email is being created, it was failing stating that the email already exists, so I made it unique using date time ticks.
5. Added unit test coverage and test.ps1 to run the unit tests with coverage (coverlet.msbuild) see coverage below
5. Updated sample application and removed all references to AspNet 2.2
6. Added project reference instead of nuget package for unit test and sample application
7. Fixed appsettings.json warning, replaced false with empty object.
8. Some files have reordered the using statements due to a plugin that I am running in vscode that will order the using clauses.
9. Updated nuspec file to include reference to Microsoft.Extensions.Identity.Core
10. Update nuspec file to version 3.1 to match aspnetcore3.1, this is just an idea...there are potentially breaking changes so a major version might be a wise idea.

Current coverage result:
``` text
+---------------------------------+--------+--------+--------+
| Module                          | Line   | Branch | Method |
+---------------------------------+--------+--------+--------+
| AspNetCore.Identity.MongoDbCore | 53.16% | 48.3%  | 58.75% |
+---------------------------------+--------+--------+--------+

+---------+--------+--------+--------+
|         | Line   | Branch | Method |
+---------+--------+--------+--------+
| Total   | 53.16% | 48.3%  | 58.75% |
+---------+--------+--------+--------+
| Average | 53.16% | 48.3%  | 58.75% |
+---------+--------+--------+--------+
```

Let me know if you want any further changes.

Cheers!

David

### CODE REVIEW

**Review by alexandre-spieser — COMMENTED** [review]
Thanks for this, this help is much appreciated left a few comments :) 
Agree with the major version as it's a breaking change.

File: sample/MongoIdentitySample.Mvc/Startup.cs [hunk 1]
```diff
@@ -71,7 +70,6 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env, ILoggerF
             if (env.IsDevelopment())
             {
                 app.UseDeveloperExceptionPage();
-                app.UseBrowserLink();
```
[the remark this answers is not in the document]
Comment (d-barker) [reply sample/MongoIdentitySample.Mvc/Startup.cs]: I was testing a few things and then after your comment reverted, I was trying to get rid of the net 2.2 dependencies and during the testing this library was pulling in some of the dependencies, but I just retested and its all good. So happy to put back :-)

File: src/Extensions/ServiceCollectionExtension.cs [hunk 2]
```diff
@@ -74,51 +77,60 @@ private static void ValidateMongoDbSettings(MongoDbSettings mongoDbSettings)
...
-            if(mongoDbContext == null)
+            if (mongoDbContext == null)
             {
-                services.AddIdentity<TUser, TRole>()
+                builder = services.AddIdentityCore<TUser>()
+                        .AddRoles<TRole>()
                         .AddMongoDbStores<TUser, TRole, TKey>(
                             mongoDbIdentityConfiguration.MongoDbSettings.ConnectionString,
-                            mongoDbIdentityConfiguration.MongoDbSettings.DatabaseName)
-                        .AddDefaultTokenProviders();
```
[the remark this answers is not in the document]
Comment (d-barker) [reply src/Extensions/ServiceCollectionExtension.cs:91]: This actually doesn't break as the call to AddDefaultTokenProviders is not in the library. The call to ConfigureMongoDbIdentity is where it breaks.

File: test/AspNetCore.Identity.MongoDbCore.IntegrationTests/AspNetCore.Identity.MongoDbCore.Test/MongoDbStoreTestBase.cs [hunk 3]
```diff
@@ -39,9 +39,12 @@ protected override void SetupIdentityServices(IServiceCollection services)
         {
             services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
             // configure the default type name
-            services.ConfigureMongoDbIdentity<TUser, TRole, TKey>(Container.MongoDbIdentityConfiguration, Container.MongoRepository.Context);
+            services.ConfigureMongoDbIdentity<TUser, TRole, TKey>(Container.MongoDbIdentityConfiguration, Container.MongoRepository.Context)
+                    .AddDefaultTokenProviders();
```
[the remark this answers is not in the document]
Comment (d-barker) [reply test/AspNetCore.Identity.MongoDbCore.IntegrationTests/AspNetCore.Identity.MongoDbCore.Test/MongoDbStoreTestBase.cs:43]: Yep

File: sample/MongoIdentitySample.Mvc/Startup.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexandre-spieser) [reply sample/MongoIdentitySample.Mvc/Startup.cs]: I've seen that you've put it back now, great thanks ;)

**Comment by alexandre-spieser** [comment]
@d-barker I am currently testing if everything works after the merge and it seems like I'm having an issue authenticating in the MVC sample app.
I can login with a dummy account I created, but I am not able to hit the ManageController. 
It looks like the authentication has been broken by the upgrade somehow, any thoughts?

**Pull request merged**